File: bench/ruby/collatz.rb
n = (ENV["BENCH_N"] || "30000").to_i

best = 0
(1...n).each do |start|
  m = start
  steps = 0
  while m != 1
    m = m.even? ? m / 2 : 3 * m + 1
    steps += 1
  end
  best = steps if steps > best
end

puts best

File: bench/node/collatz.js
const N = parseInt(process.env.BENCH_N || "500000", 10);

let best = 0;
for (let start = 1; start < N; start++) {
  let n = start, steps = 0;
  while (n !== 1) {
    if (n % 2 === 0) n = n / 2;
    else n = 3 * n + 1;
    steps++;
  }
  if (steps > best) best = steps;
}

console.log(best);
